nox/sessions.py

```diff
@@ -74,48 +70,57 @@ class Status(enum.Enum):
     SKIPPED = 2
 
 
-class SessionConfig(object):
-    """SessionConfig is passed into the session function defined in the
-    user's *nox.py*. The session function uses this object to configure the
-    virtualenv and tell nox which commands to run within the session.
+class Session(object):
+    """The Session object is passed into each user-defined session function.
+
+    This is your primary means for installing package and running commands in
+    your Nox session.
     """
-    def __init__(self, posargs=None):
-        self._commands = []
-        self.env = {}
+    def __init__(self, runner):
+        self._runner = runner
+
+    @property
+    def env(self):
         """A dictionary of environment variables to pass into all commands."""
-        self.virtualenv = True
-        """A boolean indicating whether or not to create a virtualenv. Defaults
-        to True."""
-        self.virtualenv_dirname = None
-        """An optional string that determines the name of the directory
-        where the virtualenv will be created. This is relative to the
-        --envdir option passed into the nox command. By default, nox will
-        generate the name based on the function signature."""
-        self.interpreter = None
-        """``None`` or a string indicating the name of the Python interpreter
-        to use in the session's virtualenv. If None, the default system
-        interpreter is used."""
-        self.posargs = posargs or []
-        """``None`` or a list of strings. This is set to any extra arguments
+        return self._runner.venv.env
+
+    @property
+    def posargs(self):
+        """This is set to any extra arguments
         passed to ``nox`` on the commandline."""
-        self.reuse_existing_virtualenv = False
-        """A boolean indicating whether to recreate or reuse the session's
-        virtualenv. If True, then any existing virtualenv will be used. This
-        can also be specified globally using the
-        ``--reuse-existing-virtualenvs`` argument when running ``nox``."""
+        return self._runner.global_config.posargs
 
-    def chdir(self, dir, debug=False):
-        """Set the working directory for any commands that run in this
-        session after this point.
+    @property
+    def virtualenv(self):
+        """The virtualenv that all commands are run in."""
+        return self._runner.venv
 
-        cd() is an alias for chdir()."""
-        self._commands.append(ChdirCommand(dir, debug=debug))
+    @property
+    def bin(self):
+        """The bin directory for the virtualenv."""
+        return self._runner.venv.bin
```

Comment:
What is this set to if there is no virtualenv? Or does it raise AttributeError?

Reply:
Raises AttributeError if `ProcessEnv` is being used instead. I think that's pretty fair.

Reply:
Agreed.